examples/typescript/external_signing.ts

```diff
@@ -0,0 +1,200 @@
+    return amount;
+  }
+
+  /**
+   * Pretends to sign from a cold wallet
+   * @param encodedTransaction an already encoded signing message
+   */
+  sign(encodedTransaction: Uint8Array): Uint8Array {
+    // Sending the full transaction as BCS encoded, allows for full text viewing of the transaction on the signer.
+    // However, this is not required, and the signer could just send the signing message.
+    const deserializer = new Deserializer(encodedTransaction);
+    const rawTransaction = RawTransaction.deserialize(deserializer);
+
+    // Some changes to make it signable, this would need more logic for fee payer or additional signers
+    // TODO: Make BCS handle any object type?
+    const transaction = { rawTransaction };
+    const signingMessage = this.aptos.sign.signingMessage({ transaction });
+
+    // Pretend that it's an external signer that only knows bytes using a raw crypto library
+    const signature = nacl.sign.detached(signingMessage, this.extractedPrivateKey.secretKey);
+
+    // Construct the authenticator with the public key for the submission
+    const authenticator = new AccountAuthenticatorEd25519(
+      this.account.publicKey as Ed25519PublicKey,
+      new Ed25519Signature(signature),
+    );
+
+    const serializer = new Serializer();
+    authenticator.serialize(serializer);
+    return serializer.toUint8Array();
+  }
+}
+
+const example = async () => {
+  console.log("This example will pretend that bob is on a separate server, and never access information from it");
+
+  // Setup the client
+  const config = new AptosConfig({ network: APTOS_NETWORK });
+  const aptos = new Aptos(config);
+
+  // Create two accounts
+  const alice = new ExternalSigner("Alice", ALICE_INITIAL_BALANCE);
+  const bob = new ExternalSigner("Bob", BOB_INITIAL_BALANCE);
+
+  console.log("\n=== Funding accounts ===\n");
+  await alice.setup();
+  await bob.setup();
+
+  // Show the balances
+  console.log("\n=== Balances ===\n");
+  const aliceBalance = await alice.balance();
+  const bobBalance = await bob.balance();
+
+  if (aliceBalance !== ALICE_INITIAL_BALANCE) throw new Error("Alice's balance is incorrect");
+  if (bobBalance !== BOB_INITIAL_BALANCE) throw new Error("Bob's balance is incorrect");
+
+  // Transfer between users
+  const signleSignerTransaction = await aptos.build.transaction({
+    sender: alice.address(),
+    data: {
+      function: "0x1::coin::transfer",
+      typeArguments: [APTOS_COIN],
+      functionArguments: [bob.address(), TRANSFER_AMOUNT],
+    },
+  });
+
+  // Send the transaction to external signer to sign
+  const serializer = new Serializer();
+  signleSignerTransaction.rawTransaction.serialize(serializer);
+  const rawTransactionBytes = serializer.toUint8Array();
+
+  // We're going to pretend that the network call is just an external function call
+  console.log("\n=== Signing ===\n");
+  const authenticatorBytes = alice.sign(rawTransactionBytes);
```

Comment:
I hid the account in the external signer because, that's where you'd have to get the public key and address association.

I'll move the "hot" wallet outside of it and give more concrete names